File: Classes/Expressions/Operations/AJRExponentOperator.m

#import "AJRExponentOperator.h"

@implementation AJRExponentOperator

- (id)performOperatorWithLeft:(id)left andRight:(id)right error:(NSError **)error
{
    return [NSNumber numberWithDouble:pow([left doubleValue], [right doubleValue])];
}

@end

File: Classes/Expressions/Operations/AJRExponentOperator.swift
//
//  ExponentOperator.swift
//  radar-core
//
//  Created by Alex Raftis on 8/13/18.
//

import Foundation

@objcMembers
open class AJRExponentOperator : AJROperator, AJRFloatingPointOperator, AJRIntegerOperator {
    
    public func performIntegerOperator(left: Int, right: Int) throws -> Any? {
        return Int(pow(Double(left), Double(right)))
    }
    
    public func performFloatingPointOperator(left: Double, right: Double) throws -> Any? {
        return pow(left, right)
    }
    
}
